sql: TestDistSQLFlowsVirtualTables failed

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4162676&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4162676&tab=artifacts#/) on master @ [fb72381cab272a4f03871032cc2195e97c11260a](https://github.com/cockroachdb/cockroach/commits/fb72381cab272a4f03871032cc2195e97c11260a):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3497067877
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3497067877
--- FAIL: TestDistSQLFlowsVirtualTables (11.02s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:688: unexpected output from cluster_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4164198&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4164198&tab=artifacts#/) on master @ [e8a0b75e227bf2b07207383f1ed7173da8321538](https://github.com/cockroachdb/cockroach/commits/e8a0b75e227bf2b07207383f1ed7173da8321538):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3697727178
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3697727178
--- FAIL: TestDistSQLFlowsVirtualTables (11.54s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:688: unexpected output from cluster_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4172255&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4172255&tab=artifacts#/) on master @ [58ceac139a7e83052171121b28026a7366f16f7e](https://github.com/cockroachdb/cockroach/commits/58ceac139a7e83052171121b28026a7366f16f7e):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3306148812
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3306148812
--- FAIL: TestDistSQLFlowsVirtualTables (11.65s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:688: unexpected output from cluster_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4174923&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4174923&tab=artifacts#/) on master @ [4cb2efda85aa56f1c42129fef21c2a46b8465291](https://github.com/cockroachdb/cockroach/commits/4cb2efda85aa56f1c42129fef21c2a46b8465291):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables2708899069
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables2708899069
--- FAIL: TestDistSQLFlowsVirtualTables (10.35s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:702: unexpected output from node_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.43s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4177640&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4177640&tab=artifacts#/) on master @ [6264a82d276489bc11ad87bc98fd237319501c9f](https://github.com/cockroachdb/cockroach/commits/6264a82d276489bc11ad87bc98fd237319501c9f):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1831888122
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1831888122
--- FAIL: TestDistSQLFlowsVirtualTables (10.85s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:702: unexpected output from node_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4179193&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4179193&tab=artifacts#/) on master @ [22f45e597c8f698a556f2ae32ff68e4c0327e97a](https://github.com/cockroachdb/cockroach/commits/22f45e597c8f698a556f2ae32ff68e4c0327e97a):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3927001344
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3927001344
--- FAIL: TestDistSQLFlowsVirtualTables (11.14s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:688: unexpected output from cluster_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.50s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4192615&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4192615&tab=artifacts#/) on master @ [824ee6a735f0cd2e693c2eafcaf4e5239596b31f](https://github.com/cockroachdb/cockroach/commits/824ee6a735f0cd2e693c2eafcaf4e5239596b31f):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1928909938
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1928909938
--- FAIL: TestDistSQLFlowsVirtualTables (11.05s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:702: unexpected output from node_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4206684&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4206684&tab=artifacts#/) on master @ [81c447ddb47e33d66f59acbcb1f76c37f5a148f6](https://github.com/cockroachdb/cockroach/commits/81c447ddb47e33d66f59acbcb1f76c37f5a148f6):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3907735073
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables3907735073
--- FAIL: TestDistSQLFlowsVirtualTables (10.28s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:702: unexpected output from node_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4208251&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4208251&tab=artifacts#/) on master @ [e27baa02ee9abc50d4445876e88c07181be77b55](https://github.com/cockroachdb/cockroach/commits/e27baa02ee9abc50d4445876e88c07181be77b55):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1650312602
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1650312602
--- FAIL: TestDistSQLFlowsVirtualTables (12.47s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:688: unexpected output from cluster_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.81s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4214677&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4214677&tab=artifacts#/) on master @ [ae101ea32d99a9142a319b0a1f6850ee76d55cd9](https://github.com/cockroachdb/cockroach/commits/ae101ea32d99a9142a319b0a1f6850ee76d55cd9):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1354185304
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables1354185304
--- FAIL: TestDistSQLFlowsVirtualTables (10.04s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:702: unexpected output from node_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4229167&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4229167&tab=artifacts#/) on master @ [819cc5e7a0642ebbb2872541cd227c79c5025d75](https://github.com/cockroachdb/cockroach/commits/819cc5e7a0642ebbb2872541cd227c79c5025d75):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables4104142260
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables4104142260
--- FAIL: TestDistSQLFlowsVirtualTables (9.90s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:688: unexpected output from cluster_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.40s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDistSQLFlowsVirtualTables [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4231329&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4231329&tab=artifacts#/) on master @ [fa93c68218af8094f24f421d233ee9d344b7303c](https://github.com/cockroachdb/cockroach/commits/fa93c68218af8094f24f421d233ee9d344b7303c):


```
=== RUN   TestDistSQLFlowsVirtualTables
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables2055223954
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/2 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1/1 and waiting for test to unblock...
    crdb_internal_test.go:550: stalling on scan at /Table/56/1 and waiting for test to unblock...
=== CONT  TestDistSQLFlowsVirtualTables
    crdb_internal_test.go:721: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestDistSQLFlowsVirtualTables2055223954
--- FAIL: TestDistSQLFlowsVirtualTables (10.33s)
=== RUN   TestDistSQLFlowsVirtualTables/MaxRunningFlows=0
    crdb_internal_test.go:641: waiting for remote flows to be scheduled or run
    crdb_internal_test.go:659: checking the virtual tables
    crdb_internal_test.go:702: unexpected output from node_distsql_flows on node 3
    --- FAIL: TestDistSQLFlowsVirtualTables/MaxRunningFlows=0 (1.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSQLFlowsVirtualTables.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/crdb_internal_test.go
+++ b/pkg/sql/crdb_internal_test.go
@@ -41,7 +41,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/tests"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
@@ -517,7 +516,6 @@ UPDATE system.namespace SET id = 12345 WHERE id = %d;
 
 func TestDistSQLFlowsVirtualTables(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.UnderRaceWithIssue(t, 75208, "flaky test under race")
 	defer log.Scope(t).Close(t)
 
 	const numNodes = 3
@@ -686,8 +684,18 @@ func TestDistSQLFlowsVirtualTables(t *testing.T) {
 				if maxRunningFlows == 1 {
 					expRunning, expQueued = expQueued, expRunning
 				}
-				if getNum(db, clusterScope, runningStatus) != expRunning || getNum(db, clusterScope, queuedStatus) != expQueued {
-					t.Fatalf("unexpected output from cluster_distsql_flows on node %d", nodeID+1)
+				gotRunning, gotQueued := getNum(db, clusterScope, runningStatus), getNum(db, clusterScope, queuedStatus)
+				if gotRunning != expRunning {
+					t.Fatalf("unexpected output from cluster_distsql_flows on node %d (running=%d)", nodeID+1, gotRunning)
+				}
+				if maxRunningFlows == 1 {
+					if gotQueued != expQueued {
+						t.Fatalf("unexpected output from cluster_distsql_flows on node %d (queued=%d)", nodeID+1, gotQueued)
+					}
+				} else {
+					if gotQueued > expQueued { // it's possible for the query to have already errored out
+						t.Fatalf("unexpected output from cluster_distsql_flows on node %d (queued=%d)", nodeID+1, gotQueued)
+					}
 				}
 
 				// Check node level table.
@@ -700,8 +708,18 @@ func TestDistSQLFlowsVirtualTables(t *testing.T) {
 					if maxRunningFlows == 1 {
 						expRunning, expQueued = expQueued, expRunning
 					}
-					if getNum(db, nodeScope, runningStatus) != expRunning || getNum(db, nodeScope, queuedStatus) != expQueued {
-						t.Fatalf("unexpected output from node_distsql_flows on node %d", nodeID+1)
+					gotRunning, gotQueued = getNum(db, nodeScope, runningStatus), getNum(db, nodeScope, queuedStatus)
+					if gotRunning != expRunning {
+						t.Fatalf("unexpected output from node_distsql_flows on node %d (running=%d)", nodeID+1, gotRunning)
+					}
+					if maxRunningFlows == 1 {
+						if gotQueued != expQueued {
+							t.Fatalf("unexpected output from node_distsql_flows on node %d (queued=%d)", nodeID+1, gotQueued)
+						}
+					} else {
+						if gotQueued > expQueued { // it's possible for the query to have already errored out
+							t.Fatalf("unexpected output from node_distsql_flows on node %d (queued=%d)", nodeID+1, gotQueued)
+						}
 					}
 				}
 			}

